bin-prot/src/value/layout/traverse.rs

```diff
@@ -62,23 +47,18 @@ pub enum Error {
     InvalidBranchIndex { max: usize, got: usize },
 }
 
-impl BranchingIterator for BinProtRuleIterator {
+impl Iterator for BinProtRuleIterator {
```

Comment:
Nice to see that we can use the builtin `Iterator` now. Also do we need an update to the module level docs above since this impl has changed slightly?